mlir-clang/mlir-clang.cc

```diff
@@ -459,7 +459,9 @@ int main(int argc, char **argv) {
       optPM.addPass(mlir::createCanonicalizerPass());
       optPM.addPass(mlir::createCSEPass());
 
-      pm.addPass(mlir::createInlinerPass());
+      // Disable inlining for -O0
+      if (!Opt0)
```

Comment:
Technically O0 is assumed by clang if no opts are set, can you also emulate that behavior here?

Reply:
Not quite, the condition should be inlining is enabled only of Opt1 Opt2 or Opt3 are set. Thus if O0 is set, or no opt is set the same result should come out (not setting an opt should be equivalent to O0)

Reply:
I see, that makes sense now, thanks for explaining! Will try to add another patch later.